web/middlewares/ie_test.go

```diff
@@ -0,0 +1,45 @@
+package middlewares
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/cozy/cozy-stack/pkg/instance"
+	"github.com/cozy/echo"
+	"github.com/stretchr/testify/assert"
+)
+
+var ins *instance.Instance
+
+type stupidRenderer struct{}
+
+func (sr *stupidRenderer) Render(w io.Writer, name string, data interface{}, c echo.Context) error {
+	return nil
+}
+func TestUserAgent(t *testing.T) {
+	// middleware instance
+	e := echo.New()
+
+	e.Renderer = &stupidRenderer{}
+
+	req, _ := http.NewRequest(echo.GET, "http://cozy.local", nil)
+	req.Header.Set(echo.HeaderAccept, "text/html")
+	req.Header.Set("User-Agent", "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:63.0) Gecko/20100101 Firefox/63.0") // Firefox
+
+	ins = &instance.Instance{Domain: "cozy.local", Locale: "en"}
+
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	c.Set("instance", ins)
+
+	h := CheckIE(echo.NotFoundHandler)
+	err := h(c)
+	assert.NotEqual(t, err, nil)
```

Comment:
[nitpick] You can use `assert.Error` and `assert.NoError` to make your tests clearer